sympy/printing/octave.py

```diff
@@ -468,17 +468,21 @@ def _print_Piecewise(self, expr):
                              "some condition.")
         lines = []
         if self._settings["inline"]:
-            # Express each (cond, expr) pair in a nested Horner form:
-            #   (condition) .* (expr) + (not cond) .* (<others>)
-            # Expressions that result in multiple statements won't work here.
-            ecpairs = ["({0}).*({1}) + (~({0})).*(".format
-                       (self._print(c), self._print(e))
-                       for e, c in expr.args[:-1]]
-            elast = "%s" % self._print(expr.args[-1].expr)
-            pw = " ...\n".join(ecpairs) + elast + ")"*len(ecpairs)
-            # Note: current need these outer brackets for 2*pw.  Would be
+            # This method uses the concept of vectors in octave
+            # All the expressions are stored in expression_vector
+            # All the conditions are stored in condition_vector
+            # find((condition_vector == 1)(1)) calculates the index of condition
+            # which is true. The correct expression is stored in the index of
+            # expression_vector found in the previous step
+            expression_vector = ""
+            condition_vector = ""
+            for t in expr.args:
+                expression_vector += "{0};".format(t[0])
```

Comment:
I see semicolons.  Are you building an array?  If so, how can the resulting formula be vectorized?  Or is it a cell array?